src/Microsoft.TemplateEngine.Cli/TemplateInvocationAndAcquisitionCoordinator.cs

```diff
@@ -21,71 +15,49 @@ internal class TemplateInvocationAndAcquisitionCoordinator
-        private readonly string _defaultLanguage;
         private readonly string _commandName;
         private readonly Func<string> _inputGetter;
         private readonly New3Callbacks _callbacks;
 
-        private bool _resolutionResultInitialized = false;
-        TemplateResolutionResult _templateResolutionResult;
-        ITemplateMatchInfo _templateToInvoke;
-        Status _singleMatchStatus;
-
-        public TemplateInvocationAndAcquisitionCoordinator(SettingsLoader settingsLoader, INewCommandInput commandInput, TemplateCreator templateCreator, IHostSpecificDataLoader hostDataLoader, ITelemetryLogger telemetryLogger, string defaultLanguage, string commandName, Func<string> inputGetter, New3Callbacks callbacks)
+        public TemplateInvocationAndAcquisitionCoordinator(SettingsLoader settingsLoader, INewCommandInput commandInput, ITelemetryLogger telemetryLogger,  string commandName, Func<string> inputGetter, New3Callbacks callbacks)
         {
             _settingsLoader = settingsLoader;
             _environment = _settingsLoader.EnvironmentSettings;
             _commandInput = commandInput;
-            _templateCreator = templateCreator;
-            _hostDataLoader = hostDataLoader;
             _telemetryLogger = telemetryLogger;
-            _defaultLanguage = defaultLanguage;
             _commandName = commandName;
             _inputGetter = inputGetter;
             _callbacks = callbacks;
         }
 
-        public async Task<CreationResultStatus> CoordinateInvocationOrAcquisitionAsync()
+        public async Task<CreationResultStatus> CoordinateInvocationOrAcquisitionAsync(ITemplateMatchInfo templateToInvoke)
         {
-            EnsureTemplateResolutionResult();
-
-            if (_templateToInvoke != null)
-            {
-                // invoke and then check for updates
-                CreationResultStatus creationResult = await InvokeTemplateAsync();
-                // check for updates on this template (pack)
-                await CheckForTemplateUpdateAsync();
-
-                return creationResult;
-            }
-            else
-            {
-                TemplateListResolutionResult listingTemplateListResolutionResult = TemplateResolver.GetTemplateResolutionResultForListOrHelp(_settingsLoader.UserTemplateCache.TemplateInfo, _hostDataLoader, _commandInput, _defaultLanguage);
-                return HelpForTemplateResolution.CoordinateHelpAndUsageDisplay(listingTemplateListResolutionResult, _environment, _commandInput, _hostDataLoader, _telemetryLogger, _templateCreator, _defaultLanguage, false);
-            }
+            // invoke and then check for updates
+            CreationResultStatus creationResult = await InvokeTemplateAsync(templateToInvoke).ConfigureAwait(false);
+            // check for updates on this template (pack)
+            await CheckForTemplateUpdateAsync(templateToInvoke).ConfigureAwait(false);
+            return creationResult;
```

Comment:
likely I lack scope on this.  My question...
- `creationResult` is instantiated from the result of an awaitable
- `CheckForTemplateUpdateAsync` is then called & awaited
- `creationResult` is then returned
Why is `CheckForTemplateUpdateAsync` needed to return `creationResult`?

Reply:
We will rework this part in separate PR. I will create the issue

Reply:
https://github.com/dotnet/templating/issues/2717